src/timeresp.jl

```diff
@@ -36,6 +37,9 @@ Base.step(sys::TransferFunction, t::AbstractVector; kwargs...) = step(ss(sys), t
 Calculate the impulse response of system `sys`. If the final time `Tf` or time
 vector `t` is not provided, one is calculated based on the system pole
 locations.
+The the impulse is such that the area under it is 1, with the shortest possible time-step.
+For continuos systems this is the Dirac delta function.
+For discrete systems this is the Kronecker delta function.
```

Comment:
The statement about that the impulse is called Dirac in one case and Kronecker in the other case doesn't contribute much to the understanding of the function.

Also, it is not quite the Kronecker delta but 1/Ts * \delta[k]

I absolutely agree that some info about the scaling for discrete-time systems is lacking. Something along the lines of the following is perhaps slightly easier to parse.

```suggestion
For discrete-time systems, the magnitude of the impulse is `1/Ts`.
```

Reply:
I agree that my formulation is not great. But I would not understand the impulse as well as I currently do with your suggestion, and I like understanding it more than your suggestion would tell me.

Here are two replacements I am happy with, where I prefer the first one:
"The impulse is such that the area under it is 1, with the shortest possible time-step."
I find this formulation extremely clear, consise and general for both discrete and continuous systems.

"For continuous systems the impulse is the Dirac delta function. 
For discrete functions, the impulse has width `Ts` and magnitude `1/Ts`."
I find this explanation detailed, yet short and easy to understand.
Maybe magnitude should be height, as it is more descriptive because it is positive.

Ofc feel free to rephrase it if you don't like the two options I am happy with ^_^

Reply:
I have been making a mess in a couple of commits just now, but I have made some consistensy and capitilization changes, and I also went ahead and changed the docs for impulse to my first and favourite suggestion. Ofc, that is not because this is settled, but because I had messed it up, and in fixing it I just applied the change. Still waiting to hear your opinion before changing docs for impulseplot to the same wording

Reply:
Discrete-time signals are just sequences of numbers. It is clear that a discrete-time impulse is just [a_0, 0, 0, ....].
There are no such things as area or width of discrete-time impulses, so your suggestions are a bit problematic, and I would prefer my suggestion.

The way that Matlab (and this toolbox) plots discrete-time time-responses is technically incorrect, and is probably why you would like a different formulation?

@mfalt @baggepinnen We should at least have an option for stem plots for discrete-time time responses, but I think it is better if the current approach is the default, otherwise there will be a lot of clutter.

Reply:
- The user can always supply `seriesstyle = :stem` (or `:sticks` or whatever the keyword is) if we have not forced `:step`. 
- I think with "area equal to 1" it's ment that under a ZoH assumption, the area of the impulse will be 1. We do not really have many facilities to handle different forms of discretizations in the toolbox, other than `c2d` performing ZoH-sampling.

Reply:
> The user can always supply seriesstyle = :stem (or :sticks or whatever the keyword is) if we have not forced :step.

Good to hear!

> I think with "area equal to 1" it's ment that under a ZoH assumption, the area of the impulse will be 1. We do not really have many facilities to handle different forms of discretizations in the toolbox, other than c2d performing ZoH-sampling.

Yes, you can talk about area/width of the ZoH of a discrete-time impulse. But implying that a discrete time-impulse has an area or width simply does not make sense to me. It could be that the discrete-time system represents an inherently discrete process.

Reply:
@olof3 After reading your comment a second or third time, I am realizing why my formulation is inaccurate. I am just not used to thing about discrete signals yet, at all. But you are absolutely right.


So how about something like this?

The signal is such that  `timestep * magnitude = 1`. For continuous signals, `timestep → 0`. For discrete signals, `timestep = sampleperiod = Ts`

Reply:
I think you are right that we should describe what sort of impulses that we are talking about, even if it should be quite obvious. How about the following,


```suggestion
The response to
* A unit Dirac impulse at time `t=0` (continuous-time systems)
* An impulse with magnitude `1/Ts` at time `k=0` (discrete-time systems)
```

Reply:
I have been waiting for your reply for the last 10 days, only to see you made it 8 days ago. Whoopsie

I like it a lot. My only problem is the formulation "at time `k=0`". I am in this deep enough that I am committing to the documentation, but I still don't know the variable k. Nor am I able to find it as an argument to the function. Why not call it t, if it is just some arbitrary time? I would personally find it less confusing that way.

Reply:
> even if it should be quite obvious.

As a newbie, I had NO idea what to expect a short while ago. Nothing was obvious (Picture confused student in a new course - certain about just about nothing in the new subject). Maybe a step function is something obvious, but no fancy math like a Dirac delta function. So I really like that we are making the documentation navigatable to someone who is just learning about this, and try to use Julia for it ^_^

Reply:
> I have been waiting for your reply for the last 10 days, only to see you made it 8 days ago. Whoopsie
> 
> I like it a lot. My only problem is the formulation "at time `k=0`". I am in this deep enough that I am committing to the documentation, but I still don't know the variable k. Nor am I able to find it as an argument to the function. Why not call it t, if it is just some arbitrary time? I would personally find it less confusing that way.

I see the same problem as you do, however, it it the terminology and notation used in "Computer Controlled systems", which the toolbox is aligned with to a large extent. I find it quite weird to denote discrete time with `t`. I was debating whether it should be "sample instant `k`", but went with time in accordance with (Computer Controlled Systems), it is quite clear from the context. To me the proposal seems sufficiently consistent with the literature.

@baggepinnen @mfalt we should have some document where we try to document the preferred terminology and notation to improve the consistency in the documentation. Also, any opinions for this particular case?

> As a newbie, I had NO idea what to expect a short while ago. Nothing was obvious (Picture confused student in a new course - certain about just about nothing in the new subject). Maybe a step function is something obvious, but no fancy math like a Dirac delta function. So I really like that we are making the documentation navigatable to someone who is just learning about this, and try to use Julia for it ^_^

Some things are better explained in textbooks, but a little bit of extra clarity in the docs doesn't hurt.